File: lang/go/src/fun/vac/algorithms/sorting/InsertionSort/InsertionSort.go
package InsertionSort

func Sort(arr []int) {
    var target int

    var cursor int

    for i, length := 1, len(arr); i < length; i++ {
        target = arr[i]
        for cursor = i - 1; cursor >= 0; cursor-- {
            if arr[cursor] > target {
                arr[cursor + 1] = arr[cursor]
            } else {
                break
            }
        }
        arr[cursor + 1] = target
    }
}

File: lang/ruby/src/fun/vac/algorithms/sorting/InsertionSort.rb
module InsertionSort

    def self.sort(arr)
        target = nil

        cursor = 0

        (1 ... arr.length).each do |i|
            target = arr[i]
            cursor = i
            while cursor.positive?
                if arr[cursor - 1] <= target
                    break
                end
                arr[cursor] = arr[cursor - 1]
                cursor -= 1
            end
            arr[cursor] = target
        end
    end
end
